pkg/cloudprovider/azure/block_store.go

```diff
@@ -221,6 +217,28 @@ func (b *blockStore) CreateSnapshot(volumeID, volumeAZ string, tags map[string]s
 	return snapshotName, nil
 }
 
+func getSnapshotTags(arkTags map[string]string, diskTags *map[string]*string) *map[string]*string {
+	// copy tags from disk to snapshot
```

Comment:
This technically isn't a copy, since it assigns snapshotTags to be the same as diskTags if diskTags isn't nil. Do you want to be mutating diskTags?

Reply:
it doesn't really matter here since we're not updating the disk, but easy enough to create a new map. I'll do that for clarity.